app/renderer/components/tabs/content/newSessionIcon.js

```diff
@@ -37,6 +43,29 @@ class NewSessionIcon extends React.Component {
     return props
   }
 
+  componentDidMount (props) {
+    this.transitionInIfRequired()
+  }
+
+  componentDidUpdate (prevProps) {
+    this.transitionInIfRequired(prevProps)
+  }
+
+  transitionInIfRequired (prevProps) {
+    // transition only if showPartitionIcon is newly true
+    const shouldTransitionIn = this.element && !this.props.isPinned && this.props.showPartitionIcon && this.props.partitionNumber !== 0 && (!prevProps || !prevProps.showParitionIcon || this.props.partitionNumber === 0)
```

Comment:
same as above, given `shouldTransitionIn` is shared across files, a helper method would be a nice move

Reply:
like inside `tabUIState.js`